src/bindgen/transparent_types.rs

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::collections::HashMap;
+
+use crate::bindgen::ir::{GenericParams, ItemMap, Path, Struct, Type, Typedef};
+
+/// Keeps track of named types that have the same underlying representation as some other type.
+/// This happens via `#[repr(transparent)]` structs and via typedefs.
+#[derive(Default)]
+pub struct TransparentTypes {
+    transparent: HashMap<Path, (Type, GenericParams)>,
```

Comment:
So it is quite a bit unfortunate to have to go through all items again creating this data structure when we already have a `<Path, Struct>` etc hashmap in `ItemMap`, it just feels like a lot of upfront work for something that could be much more efficient.

Reply:
Yeah, I definitely agree.  At first I tried passing `&ItemMap<Struct>` and `&ItemMap<Typedef>` references in to the `simplify_standard_types` methods, but that runs afoul of the borrow checker, since we're calling them from within a `for_all_items_mut` loop that is updating the contents of those `ItemMap`s.  Do you have any suggestions for how to break that double-borrow?